import_and_export_data/guides/create-connector.rst

```diff
@@ -315,3 +315,70 @@ If you don't see your changes, make sure you have run (``bin/console assets:inst
+
+1. You need to make sure your custom job implements ``Akeneo\Tool\Component\Batch\Job\StoppableJobInterface`` and set the last property of the constructor to "true". See the following job service definition example:
+
+.. code-block:: yaml
+    :linenos:
+
+    pimee_catalog_rule.job.xlsx_product_import_with_rules:
+        class: '%pim_connector.job.simple_job.class%'
+        arguments:
+            - '%pimee_catalog_rule.job_name.xlsx_product_import_with_rules%'
+            - '@event_dispatcher'
+            - '@akeneo_batch.job_repository'
+            -
+                - '@pim_connector.step.charset_validator'
+                - '@pimee_catalog_rule.step.xlsx_product.import'
+                - '@pimee_catalog_rule.step.xlsx_product.import_associations'
+                - '@pimee_catalog_rule.step.xlsx_product.execute_rules'
+            - true # <-- This property should be true if the job is stoppable
+        tags:
+            - { name: akeneo_batch.job, connector: '%pim_connector.connector_name.xlsx%', type: '%pim_connector.job.import_type%' }
+
+2. If your job uses tasklets (by implementing ``TaskletInterface``) to execute your business logic. You can inject the service `"akeneo_batch.job.job_stopper"` directly in your custom tasklets and use the function `JobStopper::isStopping` to know if user asked to stop the job. If this function return true, you should use the function `JobStopper::stop` and exit your Tasklet.
+
+ .. warning:: 
+ 
+Make sure that for large set of items to process, you should not call `JobStopper::isStopped` function at each time you processed an item, but rather 100 by 100. 
+
+Track the progress of a job
+---------------------------
+
+As of now, it is possible your users are not able to see the progress of your job in the UI.
+
+.. note::
+
+    If your job is composed of an ``Akeneo\Tool\Component\Batch\Step\ItemStep`` instance, as well as a reader implementing ``Akeneo\Tool\Component\Batch\Item\TrackableItemReaderInterface`` the tracking of your job should already be available in the UI.
+
+
+If your job uses a custom reader, make sure it implements ``Akeneo\Tool\Component\Batch\Item\TrackableItemReaderInterface`` and exposes the total of number that will be processed during the execution of the step.
+
+If your job uses a custom tasklet, we need to make sure of a few additional things:
+
+- your tasklet should implement the ``Akeneo\Tool\Component\Batch\Item\TrackableTaskletInterface`` interface
+- at the very beginning of the execution of the tasklet, you need to provide the step execution with the total items your tasklet will process through the ``Akeneo\Tool\Component\Batch\Model\StepExecution::setTotalItems`` function.
+- during the process of the tasklet, you need to provide the step execution with the progression the tasklet by incrementing a counter through the ``Akeneo\Tool\Component\Batch\Model\StepExecution::incrementProcessedItems`` function.
```

Comment:
maybe a small example would help here

Reply:
I agree! Let me write a php class here that does the job of a simple tasklet!

Reply:
done